src/components/Timeline/Timeline.tsx

```diff
@@ -65,6 +65,21 @@ export const Timeline: FC<TimelineProps> = ({
       setCurrentPosition(clampedValue);
       onPositionChange?.(clampedValue);
     }
+
+    setSeekViewPosition(newPosition);
+  };
+
+  const setSeekViewPosition = (newPosition: number) => {
+    const increase = newPosition > currentPosition;
+    const isInView = newPosition >= seekOffset && newPosition <= (seekOffset + seekVisibleWidth);
+
+    if (isInView) return;
+
+    if (increase && newPosition > (seekOffset + seekVisibleWidth)) {
```

Comment:
if `currentPosition` is in current view then the only check we need is `if (increase)`